File: GXpert/GXpert.Web/Modules/ServerTypes/Schools/SchoolStudentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SchoolStudentRow } from "./SchoolStudentRow";

export interface SchoolStudentColumns {
    Id: Column<SchoolStudentRow>;
    StudentPrn: Column<SchoolStudentRow>;
    SchoolName: Column<SchoolStudentRow>;
    ClassTitle: Column<SchoolStudentRow>;
    Division: Column<SchoolStudentRow>;
    RollNumber: Column<SchoolStudentRow>;
    Description: Column<SchoolStudentRow>;
    AcademicYearName: Column<SchoolStudentRow>;
}

export class SchoolStudentColumns extends ColumnsBase<SchoolStudentRow> {
    static readonly columnsKey = 'Schools.SchoolStudent';
    static readonly Fields = fieldsProxy<SchoolStudentColumns>();
}

File: GXpert/GXpert.Web/Modules/Schools/SchoolStudent/SchoolStudentColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Schools.Columns;

[ColumnsScript("Schools.SchoolStudent")]
[BasedOnRow(typeof(SchoolStudentRow), CheckNames = true)]
public class SchoolStudentColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string StudentPrn { get; set; }
    public string SchoolName { get; set; }
    public string ClassTitle { get; set; }
    [EditLink]
    public string Division { get; set; }
    public string RollNumber { get; set; }
    public string Description { get; set; }
    public string AcademicYearName { get; set; }
}